### DISCUSSION TITLE
SydneyQt - 基于InterestingDarkness的gui.py修改的EdgeGPT客户端

### DISCUSSION THREAD

**Discussion by juzeon** [viopsa233/AI-testers-Forum#46] [交流]
本来想提交PR的，但一看InterestingDarkness大佬不更了，开了新项目[ChatSydney](https://github.com/viopsa233/AI-testers-Forum/discussions/42)，好吧。但相比网页版，我个人还是喜欢Qt这样原生的客户端，于是开了一个新项目继续为大佬的gui.py加功能。

项目地址：<https://github.com/juzeon/SydneyQt>

目前在原有的基础上提升，加了一些包括预定义指令、代理、建议回复、字体设置等等的功能。如果有需求的话会继续更新。

![](https://github.com/juzeon/SydneyQt/raw/master/docs/1.png)

再次感谢InterestingDarkness大佬的[gui.py](https://github.com/InterestingDarkness/EdgeGPT/blob/sydney/src/gui.py)，也欢迎大家如果喜欢用网页版的话去支持ChatSydney。

**Comment by q2408808** [comment]
大佬，问个问题，目前这种模式，有没有可能做成2个AI进行讨论的那种交互呢?

**Reply by juzeon** [reply]
因为使用的EdgeGPT模式是每次都新开对话，发送完成上下文，不需要维护持久的websocket连线，所以是可以开多个窗口运行的

**Reply by q2408808** [reply]
所以就只能互相复制是吗

**Reply by q2408808** [reply]
这个做的确实很棒，黑暗大佬的gui完全不香了。正愁着看不到markdown的代码，想着换网页版直接在上面修改内容要麻烦很多。

**Reply by juzeon** [reply]
> 所以就只能互相复制是吗

对的，目前是这样，因为这个属于比较偏的需求所以大概不会很快去实现，不过欢迎交PR

**Reply by q2408808** [reply]
好吧。我就是纯粹对AI感兴趣，才出来看看的，对编码一窍不通。就是从老外那拿了好多关于人物专家设定的prompt，都是好几千字，想着能让它们自己交流起来就好了。

**Comment by KaneBowman** [comment]
Expecting value: line 1 column 1 (char 0)这是什么意思？

**Reply by juzeon** [reply]
是程序启动时在控制台的报错的吗？如果是的话把config.json删除试试。

**Reply by KaneBowman** [reply]
在软件里报的错

**Reply by juzeon** [reply]
> 在软件里报的错

是什么情况下出现这个错误？一直都报错吗？截个图看看

**Reply by KaneBowman** [reply]
问题已经解决，错误原因：cookies.json配置错误

**Reply by juzeon** [reply]
> 问题已经解决，错误原因：cookies.json配置错误

OK

**Comment by wangxinqijian** [comment]
填入代理显示
![image](https://github.com/viopsa233/AI-testers-Forum/assets/103314584/cd75c0bf-801c-4c94-9d48-80c78a7da10c)
不填入代理显示
![image](https://github.com/viopsa233/AI-testers-Forum/assets/103314584/1f539e73-eb9c-40f6-8e84-2b0b88c12ee6)
开不开魔法都是上面这个报错

**Reply by juzeon** [reply]
cookies被cn.bing.com污染了。开美国代理，然后打开浏览器隐私模式，登录bing.com再导出cookies。

另外可以试一下不填代理用proxifier。

**Reply by wangxinqijian** [reply]
还是不行，我是直接把之前放在InterestingDarkness大佬项目中的cookie文件复制过来的，那个项目我还没删，试了下能正常对话。但是复制过来就不行，然后我又开了美国代理登了另外一个账号获取了cookies，复制过来还是第二张图的那个报错

**Reply by wangxinqijian** [reply]
我重新下载部署你两小时前更新的项目，报错：
![image](https://github.com/viopsa233/AI-testers-Forum/assets/103314584/deb6a8b5-88da-445b-ba49-0a6499bc56cb)

**Reply by viopsa233** [reply]
> 我重新下载部署你两小时前更新的项目，报错： ![image](https://user-images.githubusercontent.com/103314584/238169800-deb6a8b5-88da-445b-ba49-0a6499bc56cb.png)

运行 `pip install -r requirements.txt
`

**Reply by wangxinqijian** [reply]
> > 我重新下载部署你两小时前更新的项目，报错： ![image](https://user-images.githubusercontent.com/103314584/238169800-deb6a8b5-88da-445b-ba49-0a6499bc56cb.png)
> 
> 运行 `pip install -r requirements.txt `

没用，我第二步就是运行 pip install -r requirements.txt了

**Reply by viopsa233** [reply]
> > > 我重新下载部署你两小时前更新的项目，报错： ![image](https://user-images.githubusercontent.com/103314584/238169800-deb6a8b5-88da-445b-ba49-0a6499bc56cb.png)
> > 
> > 
> > 运行 `pip install -r requirements.txt `
> 
> 没用，我第二步就是运行 pip install -r requirements.txt了

`pip install docx2txt`,我刚刚看了requirements.txt好像没有docx2txt

**Reply by wangxinqijian** [reply]
> > > > 我重新下载部署你两小时前更新的项目，报错： ![image](https://user-images.githubusercontent.com/103314584/238169800-deb6a8b5-88da-445b-ba49-0a6499bc56cb.png)
> > > 
> > > 
> > > 运行 `pip install -r requirements.txt `
> > 
> > 
> > 没用，我第二步就是运行 pip install -r requirements.txt了
> 
> `pip install docx2txt`,我刚刚看了requirements.txt好像没有docx2txt

感谢大佬指点，这个问题解决了。不过又回到的前面那个问题Exceeded maximum allowed redirects.为什么一直会有重定向的报错呢，费解。同一个cookies文件，用InterestingDarkness大佬的项目就可以，用这个就不行，而且我看别人好像没有出现我这个问题的。

**Reply by juzeon** [reply]
这个在聊天的网络请求方面和`gui.py`唯一的不同就是在设置中如果代理不为空就加入一个EdgeGPT支持的代理参数，除此之外就没有改动了。

**Reply by wangxinqijian** [reply]
> 这个在聊天的网络请求方面和`gui.py`唯一的不同就是在设置中如果代理不为空就加入一个EdgeGPT支持的代理参数，除此之外就没有改动了。

试了各种方法还不行，大佬能说下Exceeded maximum allowed redirects.报错涉及到的配置文件是哪个吗，我去文件里面排查下。

**Reply by juzeon** [reply]
> > 这个在聊天的网络请求方面和`gui.py`唯一的不同就是在设置中如果代理不为空就加入一个EdgeGPT支持的代理参数，除此之外就没有改动了。
> 
> 试了各种方法还不行，大佬能说下Exceeded maximum allowed redirects.报错涉及到的配置文件是哪个吗，我去文件里面排查下。

这个错误是httpx包报的，应该是在EdgeGPT创建conversation时发送的HTTP请求：<https://github.com/encode/httpx/blob/master/httpx/_client.py#L960>

出现这种现象的原因可能是在国际版bing相关接口和cn.bing.com之间反复跳转。

**Reply by wangxinqijian** [reply]
> > > 这个在聊天的网络请求方面和`gui.py`唯一的不同就是在设置中如果代理不为空就加入一个EdgeGPT支持的代理参数，除此之外就没有改动了。
> > 
> > 
> > 试了各种方法还不行，大佬能说下Exceeded maximum allowed redirects.报错涉及到的配置文件是哪个吗，我去文件里面排查下。
> 
> 这个错误是httpx包报的，应该是在EdgeGPT创建conversation时发送的HTTP请求：https://github.com/encode/httpx/blob/master/httpx/_client.py#L960
> 
> 出现这种现象的原因可能是在国际版bing相关接口和cn.bing.com之间反复跳转。

吐血，我把python卸载重装了一下，就OK了

**Comment by xbzstudio** [comment]
支持

**Comment by juzeon** [comment]
近期新增的功能，包括：
- markdown渲染，方便查看代码和回复内容
- token使用计数，便于调试
- 浏览网页内容，以及打开本地pdf、docx、pptx文档内容插入到上下文
- 以及一些细节方面的优化，等等

**Comment by Alexmercerer** [comment]
检测撤回后自动回复会做吗

**Reply by juzeon** [reply]
已加，现在会在建议框显示一条建议。可以在设置中自定义。

**Reply by Alexmercerer** [reply]
> 已加，现在会在建议框显示一条建议。可以在设置中自定义。

no close frame received or sent 的出现太频繁了，99%的时候都是发不出去或者发出去没回应，只有这个端才有问题。
![image](https://github.com/viopsa233/AI-testers-Forum/assets/108423254/3d7c5d35-3820-4267-bb3a-bbb6dcd3d9ef)

然后是我小小的提议：能不能把检测撤回后自动回复的建议内容，直接模拟用户手动操作的行为自动发送出去
![image_88](https://github.com/viopsa233/AI-testers-Forum/assets/108423254/6cd5c5d3-2752-44d4-aa76-88a16c185065)

**Reply by juzeon** [reply]
> > 已加，现在会在建议框显示一条建议。可以在设置中自定义。
> 
> no close frame received or sent 的出现太频繁了，99%的时候都是发不出去或者发出去没回应，只有这个端才有问题。 ![image](https://user-images.githubusercontent.com/108423254/238203826-3d7c5d35-3820-4267-bb3a-bbb6dcd3d9ef.png)
> 
> 然后是我小小的提议：能不能把检测撤回后自动回复的建议内容，直接模拟用户手动操作的行为自动发送出去 ![image_88](https://user-images.githubusercontent.com/108423254/238203807-6cd5c5d3-2752-44d4-aa76-88a16c185065.png)

搞定了，在设置里面设一下就行。

`no close frame received or sent`的问题，可以尝试更新下EdgeGPT依赖的版本：`pip install EdgeGPT --upgrade`。

**Comment by YOCOrange** [comment]
老哥，这种情况是什么问题，要怎么解决呢
![微信图片_20230514140236](https://github.com/viopsa233/AI-testers-Forum/assets/133559525/01da38e7-9c38-4682-b87f-9bb6fdc51765)

**Reply by YOCOrange** [reply]
多谢指点，运行pip install docx2txt后就可以了

**Comment by PoseidonLi0514** [comment]
大佬能加个自动解析数学公式吗

**Reply by juzeon** [reply]
已加

**Comment by dot307** [comment]
支持！！！！！！！

**Comment by wuyujiax** [comment]
在官网Bing的输入框，可以通过F12更改”maxlength“数值实现输入超长文本,想问问是否在客户端可以实现类似功能？因为当前输入长文本会报错”received 1000 (OK); then sent 1000 (OK)“

**Reply by ghost** [reply]
什么文本官网能输入，客户端不能输入

**Reply by wuyujiax** [reply]
输入文本的tokens长度限制，在网页版能接收到差不多18000-20000个左右tokens的长文本（超了也会报错和忘记上下文），客户端的话我尝试大概是能录入10000个左右，超了提示”received 1000 (OK); then sent 1000 (OK)“。

**Reply by ghost** [reply]
长文本请输入上面的框

**Comment by q2408808** [comment]
![H17BUNF)FNQ62WEWG H~2PL](https://github.com/viopsa233/AI-testers-Forum/assets/128444814/7b7898ce-a5af-4d75-8f45-5910e656337d)
聊着l聊着就开始缩行，一行半句，甚至一个词，是什么情况啊。

**Reply by juzeon** [reply]
这是bing本身的问题，可以尝试更换prompt。

**Comment by yehuankun** [comment]
![QQ截图20230516121022](https://github.com/viopsa233/AI-testers-Forum/assets/127468400/751baa37-574c-4ec7-a75b-6d3c48c3cb55)
为什么会报错呢~不理解·~

**Reply by yehuankun** [reply]
@juzeon @

**Reply by juzeon** [reply]
`pip install -r requirements --upgrade`。若还不行，完全卸载python之后重装。

**Comment by NostalgiaGit** [comment]
感谢大佬，客户端限制能显示代码和数学公式，方便多了！不知道大佬有没有计划把超链接也做进来，现在ai引用的链接在客户端还是没法查看

**Reply by ghost** [reply]
很难啦，我当时做gui.py是有各种考虑的，要高级特性用ChatSydney

**Reply by NostalgiaGit** [reply]
> 很难啦，我当时做gui.py是有各种考虑的，要高级特性用ChatSydney

ChatSydney都能显示，很方便，但唯一问题是修改聊天记录不如之前的GUI方便，难顶:joy: 这就是鱼和熊掌不可兼得吗:joy:

**Reply by ghost** [reply]
因为ChatSydney比较高级，所以可以选择性在界面上隐藏一些消息，搜索结果的消息和搜索建议的消息
ChatSydney修改聊天记录功能，可以做，只是没动力做

**Reply by NostalgiaGit** [reply]
> 因为ChatSydney比较高级，所以可以选择性在界面上隐藏一些消息，搜索结果的消息和搜索建议的消息 ChatSydney修改聊天记录功能，可以做，只是没动力做

嗯嗯，干正事时我用ChatSydney，其实倒也不需要修改聊天记录。涩涩换GUI:satisfied:

**Reply by juzeon** [reply]
![eUS7AkWCvG](https://github.com/viopsa233/AI-testers-Forum/assets/12206799/65e7769a-462a-42b2-af5f-beb26c7dd52b)

改进了一下链接显示的格式，现在应该比之前好点。

**Reply by ghost** [reply]
从AdaptiveCard取回复有链接，但这不符合Sydney原生格式所以我没做，ChatSydney是从AdaptiveCard取回复显示在UI上

**Reply by NostalgiaGit** [reply]
多谢多谢，更新了一下，赞！

**Comment by jishux2** [comment]
请问这个是什么问题啊，今天第一次见
![bing](https://github.com/viopsa233/AI-testers-Forum/assets/120390643/3666fb7b-43b6-4888-a58f-11b0ea1d8ea0)

**Reply by juzeon** [reply]
网络问题，检查一下代理配置。

**Reply by jishux2** [reply]
可能和bing的版本有关？换了个有历史记录框的最新网站的cookies就行了，之前用的账号不知道为什么聊天界面还是旧版，没有更新保存对话记录的功能

**Comment by NostalgiaGit** [comment]
大佬请问这个报错信息是怎么回事？一旦ai开始搜索内容时就出现这个错，普通聊天没事
![Error1](https://github.com/viopsa233/AI-testers-Forum/assets/132838751/360b28a5-4c29-4d2b-9b71-df54d332a73a)
![Error2](https://github.com/viopsa233/AI-testers-Forum/assets/132838751/22c2c4ed-13de-4a80-9fdb-ed63b7d56463)

**Reply by juzeon** [reply]
已修复。这是在没有找到搜索结果的情况下，JSON解析有问题。

**Reply by NostalgiaGit** [reply]
谢谢大佬，解决了

**Comment by Viryxs** [comment]
![346](https://github.com/viopsa233/AI-testers-Forum/assets/97890666/f2f43ab7-36a5-450f-8ea7-67d2749bc522)
大佬我想问一下这个报错是为什么呀？聊着聊着突然就这样了，换了号之后发送第一句话之后也是这个错误。

**Reply by juzeon** [reply]
可能是代理问题，也可能是每天聊天次数达到上限。

**Comment by q2408808** [comment]
现在的QT客户端有生成和显示图片的功能吗？

**Reply by juzeon** [reply]
目前暂时没有

**Reply by q2408808** [reply]
不知道为什么 今天疯狂出现InternalError: Unhandled Dependency Failure: DeepLeo的错误提示这是什么情况？网络问题吗

**Reply by q2408808** [reply]
还有指定的网络名不再可用

**Reply by juzeon** [reply]
> 不知道为什么 今天疯狂出现InternalError: Unhandled Dependency Failure: DeepLeo的错误提示这是什么情况？网络问题吗

可能是网络问题，也可能是cookies问题。

现在Bing不需要登录也可以用了，所以可能cookies.json已经过期了，但客户端识别不出来，于是使用未登录身份聊天。而未登录用户有各种各样的限制。

建议重新导出下cookies。

**Comment by ghost** [comment]
https://github.com/acheong08/EdgeGPT/pull/447
提交了一个PR恢复老GPT-4平衡模式，可以对比一下老平衡模式和创造模式

**Comment by gunshipwu** [comment]
D:\SydneyQt-master>python.exe .\main.py
Traceback (most recent call last):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\streams\tls.py", line 130, in _call_sslobject_method
    result = func(*args)
             ^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\ssl.py", line 921, in read
    v = self._sslobj.read(len)
        ^^^^^^^^^^^^^^^^^^^^^^
ssl.SSLWantReadError: The operation did not complete (read) (_ssl.c:2546)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\backends\asyncio.py", line 34, in read
    return await self._stream.receive(max_bytes=max_bytes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\streams\tls.py", line 195, in receive
    data = await self._call_sslobject_method(self._ssl_object.read, max_bytes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\streams\tls.py", line 137, in _call_sslobject_method
    data = await self.transport_stream.receive()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 1265, in receive
    await self._protocol.read_event.wait()
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\asyncio\locks.py", line 213, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_exceptions.py", line 10, in map_exceptions
    yield
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\backends\asyncio.py", line 32, in read
    with anyio.fail_after(timeout):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_core\_tasks.py", line 118, in __exit__
    raise TimeoutError
TimeoutError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_transports\default.py", line 60, in map_httpcore_exceptions
    yield
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_transports\default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\connection_pool.py", line 253, in handle_async_request
    raise exc
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\connection_pool.py", line 237, in handle_async_request
    response = await connection.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\connection.py", line 90, in handle_async_request
    return await self._connection.handle_async_request(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\http11.py", line 116, in handle_async_request
    raise exc
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\http11.py", line 95, in handle_async_request
    ) = await self._receive_response_headers(**kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\http11.py", line 159, in _receive_response_headers
    event = await self._receive_event(timeout=timeout)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_async\http11.py", line 195, in _receive_event
    data = await self._network_stream.read(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\backends\asyncio.py", line 31, in read
    with map_exceptions(exc_map):
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpcore\_exceptions.py", line 14, in map_exceptions
    raise to_exc(exc)
httpcore.ReadTimeout

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\SydneyQt-master\main.py", line 298, in send_message
    chatbot = await Chatbot.create(cookies=cookies, proxy=proxy if proxy != "" else None)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\EdgeGPT.py", line 626, in create
    await _Conversation.create(self.cookies, self.proxy),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\EdgeGPT.py", line 385, in create
    response = await client.get(
               ^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1754, in get
    return await self.request(
           ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1530, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1617, in send
    response = await self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1645, in _send_handling_auth
    response = await self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1682, in _send_handling_redirects
    response = await self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_client.py", line 1719, in _send_single_request
    response = await transport.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_transports\default.py", line 352, in handle_async_request
    with map_httpcore_exceptions():
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\gunsh\AppData\Local\Programs\Python\Python311\Lib\site-packages\httpx\_transports\default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadTimeout

今天晚上回家，就连不上了，可以直接访问官方网页版newbing.
暗黑大佬原版的 ChatSydney 也连不上了，问题一样。 换号也不起作用。

但暗黑大佬早期的EdgeGPT-sydney 还是能用的。幸好当时没删。

而且用的同一个cookies.json。

不知道是微软的措施还是网络抽疯。

**Reply by YOCOrange** [reply]
同，我这里也一样

**Reply by jishux2** [reply]
挂梯子，然后这里指定proxy，https://github.com/juzeon/SydneyQt/blob/d157d2cd3ee75a9ed5a9c5c213cf6dc3c8796fe5/main.py#L292

**Reply by YOCOrange** [reply]
> 挂梯子，然后这里指定proxy，https://github.com/juzeon/SydneyQt/blob/d157d2cd3ee75a9ed5a9c5c213cf6dc3c8796fe5/main.py#L292
也就是说你那里现在还能用是吗？

**Reply by YOCOrange** [reply]
> 挂梯子，然后这里指定proxy，https://github.com/juzeon/SydneyQt/blob/d157d2cd3ee75a9ed5a9c5c213cf6dc3c8796fe5/main.py#L292

多谢指点，已经搞定了，在setting里面设置proxy为http://127.0.0.1:xxxx即可，xxxx为梯子的本地代理端口，Clash默认是7890，别的翻墙软件就不清楚了。

**Comment by gunshipwu** [comment]
大佬，官方网页版以经可以看图了，您这个项目会跟进这个功能吗？

**Reply by juzeon** [reply]
目前这个功能应该还在灰度测试，我这边还没法使用。等陆续开放了应该会更新。

**Comment by codescipio** [comment]
这支持win7吗，Dark大大的win7可以正常使用

**Comment by juzeon** [comment]
Qt的支持用的是Pyside6实现，应该是不支持win7。



---原始郵件---
寄件人： ***@***.***&gt;
傳送時間： 2023年6月23日(週五) 中午12:44
收件人： ***@***.***&gt;;
抄送： ***@***.******@***.***&gt;;
主旨： Re: [viopsa233/AI-testers-Forum] SydneyQt - 基于InterestingDarkness的gui.py修改的EdgeGPT客户端 (Discussion #46)




 
这支持win7吗，Dark大大的win7可以正常使用

**Reply by codescipio** [reply]
唉……这样子啊

**Comment by xuyufenfei** [comment]
![Screenshot_2023-06-30-00-33-35-378_com termux](https://github.com/viopsa233/AI-testers-Forum/assets/133896005/3e85b388-29cf-4dc5-afdf-8385b86c5b7c)

**Reply by juzeon** [reply]
只支持Windows 10及以上版本，不支持Android。

**Reply by xuyufenfei** [reply]
好的，谢谢大佬

**Comment by Altria7** [comment]
请问让bing画画返回的链接点进去后一直显示创作中，出不来图是怎么回事呀？

**Reply by juzeon** [reply]
应该是个bug，可以先手动复制keyword去官网画。

**Comment by Yimo-Senior** [comment]
现在bing需要真人验证了
![image](https://github.com/viopsa233/AI-testers-Forum/assets/130832815/d2075684-7490-409f-a848-9a397a754bb6)

**Reply by juzeon** [reply]
去网页版聊一句然后再导出cookies。

**Reply by for3ever3** [reply]
也不用，去验证一下发再发条新的就行了

**Comment by nonAme160** [comment]
0, message='', url=URL('https://edgeservices.bing.com/edgesvc/turing/conversation/create')
这个咋解决

**Reply by juzeon** [reply]
梯子问题。

**Comment by jishux2** [comment]
从刚刚开始，似乎在SydneyQt中使用过cookies的账号都会陷入无限的人机验证中
![7TS @ 7(T( UQTHIHOA{32](https://github.com/viopsa233/AI-testers-Forum/assets/120390643/db57e3ab-6051-44e4-9c43-801b23759a32)
是因为微软加强了对第三方客户端的封控？

**Reply by juzeon** [reply]
新注册的账号也是这样吗？

**Reply by jishux2** [reply]
刚注册的客户端正常。但使用新账号的cookies，SydneyQt同样报错`CaptchaChallenge: User needs to solve CAPTCHA to continue.`，再然后去客户端验证也会变成上面这样

**Reply by juzeon** [reply]
试一下用Bing安卓客户端聊一句，我试了可以。

**Reply by jishux2** [reply]
嗯这样正常了，感谢

**Comment by Hakiana** [comment]
试了两个不同的cookie，都出现了“conversationSignature”的提示，两个cookie的账号在网页端都能正常使用NewBing。
![GZ}8K%AM` $ ZXIFKF`81IF](https://github.com/viopsa233/AI-testers-Forum/assets/133541685/df857cc8-1d2f-4e42-8585-adc3e6acef02)
![}F87NDGGLZQ~0{@1}7_6DTT](https://github.com/viopsa233/AI-testers-Forum/assets/133541685/abd04d7f-1ee4-41c1-b575-bca9a5d17169)

**Reply by saloc1984** [reply]
一样，看起来像是微软那里有改动。

**Reply by juzeon** [reply]
已修复

**Reply by Hakiana** [reply]
大感谢！

**Reply by saloc1984** [reply]
好耶！

**Comment by Altria7** [comment]
请问chatgpt是3.5还是3.0？我问它回答我是gpt3.0

**Reply by juzeon** [reply]
GPT自己的回答基于训练数据，对于版本问题而言不可信。

**Reply by Altria7** [reply]
这是3.5吧
<img width="289" alt="image" src="https://github.com/viopsa233/AI-testers-Forum/assets/95865658/6eabffbf-c899-4c47-94f1-21e67e4d33a4">

**Comment by nonAme160** [comment]
![image](https://github.com/viopsa233/AI-testers-Forum/assets/125792179/8b94297a-eeb3-4dc5-a298-5ede20bab4a0)
请问这个咋解决

**Reply by juzeon** [reply]
pip install -r requirements.txt

**Reply by nonAme160** [reply]
输入过了，所有的module都显示已经满足，运行main.py就报这个错

**Reply by juzeon** [reply]
重装一下Python最新版本。

**Comment by XianYuDaXian** [comment]
作者你好，能不能考虑下局域网web端功能，你的这个是我用下来最简单易用的，别的项目都是要么只有gpt claude要么只有newbing，唯一有整合的web Pandora又老是报错。

**Reply by juzeon** [reply]
要实现web版基本相当于把所有代码重写一遍，可以考虑用ChatSydney。这个项目就是为桌面端设计的。
